### PR TITLE
Debate fetcher

### PR DESCRIPTION
**Pull request by MihneaLungu**

### CODE REVIEW

File: ptracking/fetch-debates/insert_debate.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from fetch_debate_XML import *
+from psycopg2 import connect
+
+
+def insertDebate(conn, date_start, date_end = None):
+    curr = conn.cursor()
+    speeches = getVignettesXML(date_start, date_end)
+    try:
+        for vignette in speeches:
+            curr.execute("INSERT INTO vignette (mp_id, debate_id, content, p_id) values (%s, %s, %s, %s) ON CONFLICT DO NOTHING",
```
Comment (abulavin) [inline ptracking/fetch-debates/insert_debate.py:10]: Why `ON CONFLICT DO NOTHING`? Does this allow duplicate entries?

File: ptracking/fetch-debates/insert_debate.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from fetch_debate_XML import *
+from psycopg2 import connect
+
+
+def insertDebate(conn, date_start, date_end = None):
```
Comment (abulavin) [inline ptracking/fetch-debates/insert_debate.py:5]: If it makes it easier, you can use the `cursor` context manager to handle connections to the database for you https://github.com/abulavin/petition-tracking/blob/main/ptracking/database/database.py#L8

File: ptracking/fetch-debates/test_debate_fetcher.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        port="5432",
+        database="test",
+        user="petitions")
+
+        curr = conn.cursor()
+        curr.execute("TRUNCATE TABLE vignette")
+        insertDebate(conn, "1919-02-04", "1919-02-05")
+        curr.execute("SELECT * FROM VIGNETTE")
+        count = curr.rowcount
+        self.assertEqual(count, 411)
```
Comment (abulavin) [inline ptracking/fetch-debates/test_debate_fetcher.py:44]: What does `411` mean?

File: ptracking/fetch-debates/test_debate_fetcher.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        port="5432",
+        database="test",
+        user="petitions")
+
+        curr = conn.cursor()
+        curr.execute("TRUNCATE TABLE vignette")
+        insertDebate(conn, "1919-02-04")
+        curr.execute("SELECT * FROM VIGNETTE")
+        count = curr.rowcount
+        self.assertEqual(count, 28)
```
Comment (abulavin) [inline ptracking/fetch-debates/test_debate_fetcher.py:59]: Same here, what does `28` represent?

File: .gitignore [hunk 5]
```diff
@@ -131,4 +131,6 @@ dmypy.json
 # Pyre type checker
 .pyre/
 
-.vscode
\ No newline at end of file
+.vscode
+config.yml
+config.yml
```
Comment (abulavin) [inline .gitignore:136]: I think we only need this once

**Review by abulavin — APPROVED** [review]

File: ptracking/fetch-debates/insert_debate.py (hunk 1 above)
Comment (MihneaLungu) [reply ptracking/fetch-debates/insert_debate.py:10]: Nope, basically, if an entry is already in the database (based on the primary key), it does not insert it again. So vignettes already present in the database are not inserted again.

File: ptracking/fetch-debates/insert_debate.py (hunk 2 above)
Comment (MihneaLungu) [reply ptracking/fetch-debates/insert_debate.py:5]: I wanted to use that but you know that I have that "ptracking module not found" problem

File: ptracking/fetch-debates/test_debate_fetcher.py (hunk 3 above)
Comment (MihneaLungu) [reply ptracking/fetch-debates/test_debate_fetcher.py:44]: I am checking if the speeches are successfully added to the test database and I know that for those two dates there are 411 speeches in total. I make sure the table is emptied before by running truncate

File: ptracking/fetch-debates/test_debate_fetcher.py (hunk 4 above)
Comment (MihneaLungu) [reply ptracking/fetch-debates/test_debate_fetcher.py:59]: Same as above

File: .gitignore (hunk 5 above)
Comment (MihneaLungu) [reply .gitignore]: Removed the duplicate line

**Pull request merged**